src/Aspire.Hosting/ApplicationModel/ExistingResourceAnnotation.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Represents a resource that is not managed by Aspire's provisioning or
+/// container management layer.
+/// </summary>
+public class ExistingResourceAnnotation(string name, bool isPublishMode = false) : IResourceAnnotation
```

Comment:
Is this supposed to be Azure specific?

Reply:
Based on what we decided, yes. Are you indicating that the type name should indicate it's an Existing*Azure*ResourceAnnotation?

Reply:
It's in the wrong assembly.

Reply:
This should be ExistingAzureResourceAnnotation and it needs the subscription id, resource group and name as properties.

Reply:
I remove subscription ID and resource group from this model because of:

> Support for referencing existing resources in other subscription IDs/resource groups isn't implemented due to a gap in the underlying provisioning APIs. See https://github.com/Azure/azure-sdk-for-net/issues/47980. There are workarounds for the but the juice might not be worth the squeeze.

We can't materialize these into scopes in the existing reference in the Bicep identifier at the moment so it doesn't make sense to expose an API for something that isn't supported at the moment.

We can add them later easily enough but if we have them now they'd be non-functional.

Reply:
I think we should at a minimum add the resource group so that we can play with what it would look like in a custom resource even though the Azure.Provisioning libs can't directly do it yet (we can try it out with the bicep resource). The reason it's important is because in a normal aspire deployment, the resource group is created on the fly, so at a minimum, using exisiting resource would look like a name + a different resource group in the same subscription.

Reply:
> The reason it's important is because in a normal aspire deployment, the resource group is created on the fly, so at a minimum, using exisiting resource would look like a name + a different resource group in the same subscription.

Hmmm...I see. Are you cool with iterating on this in a separate PR?

Reply:
No, I think this pr is where we iterate